alembic/versions/ed775af5e086_rdiscrowd_4967_add_partial_index_for_.py

```diff
@@ -0,0 +1,27 @@
+"""RDISCROWD-4967 add partial index for task table
+
+Revision ID: ed775af5e086
+Revises: 29c0cc545d6d
+Create Date: 2022-02-28 15:01:53.582867
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'ed775af5e086'
+down_revision = '29c0cc545d6d'
+
+from alembic import op
+
+
+def upgrade():
+    op.create_index('task_state_calibration_exported_idx', 'task',
+                    ['id', 'created', 'project_id', 'state', 'quorum',
+                     'calibration', 'priority_0', 'info', 'n_answers',
+                     'fav_user_ids', 'exported', 'user_pref', 'worker_pref',
+                     'worker_filter', 'gold_answers', 'expiration'],
+                    postgresql_where="(state = 'completed'::text OR calibration = 1) AND exported = false"
+                    )
```

Comment:
Thanks for sharing stats on impact of creating index on inserts/updates. As described [here](https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#indexes-with-concurrently), there's an option to create indexes on table row without locking the table thereby allowing concurrent updates to the table. As per docs, this can be set with passing `postgresql_concurrently=True`

Reply:
Thanks for the catch. I originally thought to execute the alembic script at off hour, but I think we should make it concurrent not to affect update/insert/delete as you suggested. The tradeoff is that, with "concurrent" option, the index creation will take much longer. And there could be other caveat like building could fail(see [here](https://www.postgresql.org/docs/10/sql-createindex.html#SQL-CREATEINDEX-CONCURRENTLY)). But still worth it and we might still need to execute it at the off hour(when less update, the index creation could speed up). I will commit a change to address this. Thank you.